apisix/plugins/echo.lua

```diff
@@ -70,4 +94,22 @@ function _M.body_filter(conf, ctx)
     ngx.arg[2] = true
 end
 
+function _M.access(conf, ctx)
+    local auth_header = core.request.header(ctx, "Authorization")
+
+    if auth_header and auth_header == "userpass" then
+        return 200, "authorized body"
+    end
+    return 401, "unauthorized body"
```

Comment:
I was confused about this process. @agile6v Do you think we should keep this?

Reply:
So we talked about keeping this very simple. I thought maybe  to add the flow of basic auth plugin to extract username and PWD but we just want to convey the idea as per @agile6v. Like if someone need to access a protected resource in the access phase we need to have  some kind of an auth workflow 

WDYT @agile6v

Reply:
Hi @Akayeshmantha I mean we could specify the header and value, which is flexible and simple.

Reply:
should we drop this code? https://github.com/apache/incubator-apisix/pull/1708/commits/dcbcc69a1c334761d01d3e85bfb3d60c35b90413#diff-32537bd33cf9a44d9dd277b87d1ce5f9R100-R104

Reply:
Hi @membphis  so I pushed the latest one

Reply:
of course with more tests.